pkg/mountutil/mountutil_windows.go

```diff
@@ -16,7 +16,43 @@
 
 package mountutil
 
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containerd/containerd/oci"
+	"github.com/sirupsen/logrus"
+)
+
 func getUnprivilegedMountFlags(path string) ([]string, error) {
 	m := []string{}
 	return m, nil
 }
+
+// DefaultPropagationMode is the default propagation of mounts
+// where user doesn't specify mount propagation explicitly.
+// See also: https://github.com/moby/moby/blob/v20.10.7/volume/mounts/windows_parser.go#L440-L442
+const DefaultPropagationMode = ""
+
+// parseVolumeOptions parses specified optsRaw with using information of
+// the volume type and the src directory when necessary.
+func parseVolumeOptions(vType, src, optsRaw string) ([]string, []oci.SpecOpts, error) {
+	var writeModeRawOpts []string
+	for _, opt := range strings.Split(optsRaw, ",") {
+		switch opt {
+		case "rw":
+			writeModeRawOpts = append(writeModeRawOpts, opt)
+		case "ro":
+			writeModeRawOpts = append(writeModeRawOpts, opt)
+		default:
+			logrus.Warnf("unsupported volume option %q", opt)
+		}
```

Comment:
Sorry. We should not print warn when `opt == ""` on windows as well. I'll fix it.

Reply:
done